staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/validator.go

```diff
@@ -86,7 +85,16 @@ func (a customResourceValidator) ValidateUpdate(ctx context.Context, obj, old ru
 	var allErrs field.ErrorList
 
 	allErrs = append(allErrs, validation.ValidateObjectMetaAccessorUpdate(objAccessor, oldAccessor, field.NewPath("metadata"))...)
-	allErrs = append(allErrs, apiservervalidation.ValidateCustomResource(nil, u.UnstructuredContent(), a.schemaValidator)...)
+	if updateValidator, ok := a.schemaValidator.(apiservervalidation.SchemaUpdateValidator); ok {
```

Comment:
do we need this type-cast? Can we make every instsance support both methods? This kind of type-checks break static typing, make semantics harder.

Reply:
refactored as you suggested